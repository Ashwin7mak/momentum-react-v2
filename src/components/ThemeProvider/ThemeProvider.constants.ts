const CLASSES = {
  root: 'md-theme-provider',
};

const THEME_CLASS_PREFIX = 'md-theme';

export {
  CLASSES,
  THEME_CLASS_PREFIX,
};
