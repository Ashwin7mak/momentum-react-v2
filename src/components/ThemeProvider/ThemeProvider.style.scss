.md-theme-provider-wrapper {
  border: 0;
  height: 100%;
  margin: 0;
  padding: 0;
  width: 100%;
  color: var(--theme-text-primary-normal);
  font-family: var(--md-globals-font-default);
  font-weight: normal;

  // Firefox fix for scrollbar
  // On macOS, you need to set the General > Show scroll bars
  // setting in System Preferences to "Always" for this property to have any effect.
  scrollbar-color: var(--scrollbar-track-background) var(--scrollbar-thumb-background);

}

/* Global variables not available as tokens. */
.md-theme-provider-globals {
  --md-globals-border-clear: 0rem solid rgba(0, 0, 0, 0);
  --md-globals-focus-ring-box-shadow: 0 0 0 0.0625rem var(--theme-background-solid-primary-normal),
    0 0 0 0.125rem var(--theme-outline-focus-normal);
  --md-globals-elevation-4: 0px 24px 48px rgba(0, 0, 0, 0.2), 0px 0px 1px rgba(0, 0, 0, 0.12);
  --md-globals-focus-ring-outline: none;
  --md-globals-font-default: CiscoSansTT;
}

/* Scrollbar Theme */

/* width */
::-webkit-scrollbar {
  width: 1rem;
}

/* Track */
::-webkit-scrollbar-track {
  background-color: var(--scrollbar-track-background);
  border-radius: 100vh;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background-color: var(--scrollbar-thumb-background);
  border: 0.25rem solid transparent;
  border-radius: 0.5rem;
  background-clip: content-box;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: var(--scrollbar-thumb-background-hovered);
  border: 0.25rem solid transparent;
  border-radius: 0.5rem;
  background-clip: content-box;
}

/* Handle on press */
::-webkit-scrollbar-thumb:active {
  background: var(--scrollbar-thumb-background-pressed);
  border: 0.25rem solid transparent;
  border-radius: 0.5rem;
  height: 6.25rem;
  background-clip: content-box;
}

/* Corner */
::-webkit-scrollbar-corner {
  background: inherit;
}
