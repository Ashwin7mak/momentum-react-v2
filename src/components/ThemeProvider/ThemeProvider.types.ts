import { ReactNode } from 'react';

export type ThemeNames =
  | 'darkBronze'
  | 'darkIndigo'
  | 'darkJade'
  | 'darkLavender'
  | 'darkRose'
  | 'darkWebex'
  | 'lightBronze'
  | 'lightIndigo'
  | 'lightJade'
  | 'lightLavender'
  | 'lightRose'
  | 'lightWebex';

export interface Props {
  /**
   * The child components of this component.
   */
  children?: ReactNode;

  /**
   * The id of this component.
   */
  id?: string;

  /**
   * The theme name to provide styles from to all child components.
   */
  theme?: ThemeNames;
}
