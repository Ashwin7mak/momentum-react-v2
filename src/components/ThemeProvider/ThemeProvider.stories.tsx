import React, { FC } from 'react';
import { Story } from '@storybook/react';

// This reference will be removed once themes are available.
import './ThemeProvider.stories.css';

import ThemeProvider, { ThemeProviderProps as Props } from './';

export default {
  title: 'ThemeProvider',
  component: ThemeProvider,
};

const ExampleDiv: FC = () => (
  <div style={{
    alignItems: 'center',
    backgroundColor: 'var(--sbe-bg-color, #eee)',
    color: 'var(--sbe-font-color, #333)',
    display: 'flex',
    height: '4rem',
    justifyContent: 'center',
    width: '16rem',
  }}>Example Container</div>
);

const Template: Story<Props> = (args) => (
  <ThemeProvider {...args}>
    <ExampleDiv />
  </ThemeProvider>
);

const Themed = Template.bind({});

Themed.args = {
  theme: 'storybook-example-theme',
};

const Unthemed = Template.bind({});

Unthemed.args = {
  theme: undefined,
};

export {
  Themed,
  Unthemed,
};
