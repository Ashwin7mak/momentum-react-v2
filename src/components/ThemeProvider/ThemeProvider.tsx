import React, { FC } from 'react';

import { CLASSES, THEME_CLASS_PREFIX } from './ThemeProvider.constants';
import { Props } from './ThemeProvider.types';
import './ThemeProvider.style.css';

/**
 * Provides a collection of CSSVariables based on a ThemeToken to all child elements.
 */
const ThemeProvider: FC<Props> = ({ children, theme }: Props) => (
  <div className={`${CLASSES.root} ${THEME_CLASS_PREFIX}-${theme}`}>{children}</div>
);

export default ThemeProvider;
