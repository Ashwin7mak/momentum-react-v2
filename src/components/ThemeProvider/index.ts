import { default as ThemeProvider } from './ThemeProvider';
import { Props } from './ThemeProvider.types';

export type ThemeProviderProps = Props;

export default ThemeProvider;
