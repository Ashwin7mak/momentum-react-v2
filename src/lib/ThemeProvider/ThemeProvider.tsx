import React, { FC } from 'react';

import { CLASSES, THEME_CLASS_PREFIX } from './ThemeProvider.constants';
import { ThemeProviderProps as Props } from './ThemeProvider.types';
import './ThemeProvider.style.css';

/**
 * Provides a collection of CSSVariables based on a ThemeToken to all child elements.
 */
const ThemeProvider: FC<Props> = ({ children, token }: Props) => (
  <div className={`${CLASSES.main} ${THEME_CLASS_PREFIX}-${token}`}>{children}</div>
);

export default ThemeProvider;