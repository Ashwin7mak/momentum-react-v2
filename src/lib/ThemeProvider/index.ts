import { default as ThemeProvider } from './ThemeProvider';
import { ThemeProviderProps as Props } from './ThemeProvider.types';

export type ThemeProviderProps = Props;

export default ThemeProvider;
