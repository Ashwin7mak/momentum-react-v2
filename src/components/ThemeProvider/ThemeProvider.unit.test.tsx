import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from '@momentum-ui/react';

import { CLASSES, THEME_CLASS_PREFIX } from './ThemeProvider.constants';

describe('<ThemeProvider />', () => {
  let container;

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      container = mount(<ThemeProvider theme="test" />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its main class', () => {
      expect.assertions(1);

      const component = mount(<ThemeProvider />).childAt(0);

      expect(component.hasClass(CLASSES.root)).toBe(true);
    });

    it('should have an abstracted theme class', () => {
      expect.assertions(1);

      const theme = 'test';

      const component = mount(<ThemeProvider theme={theme} />).childAt(0);

      expect(component.hasClass(`${THEME_CLASS_PREFIX}-${theme}`)).toBe(true);
    });

    it('should mount children to the component nodes', () => {
      expect.assertions(1);

      const children = [
        <div key="0" />,
        <div key="1" />,
        <div key="2" />,
      ];

      const component = mount(<ThemeProvider>{children}</ThemeProvider>).childAt(0);

      expect(component.children().length).toBe(children.length);
    });
  });
});
