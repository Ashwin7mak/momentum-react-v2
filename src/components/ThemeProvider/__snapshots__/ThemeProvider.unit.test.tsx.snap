// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeProvider /> snapshot should match snapshot 1`] = `
<ThemeProvider
  theme="test"
>
  <div
    className="md-theme-provider md-theme-test"
  />
</ThemeProvider>
`;
